src/middlewares/passport.js

```diff
@@ -18,30 +19,64 @@
 const passport = require('passport');
 const OAuth2Strategy = require('passport-oauth2');
 const jwt = require('jsonwebtoken');
+const https = require('https');
+const config = require(process.cwd() + '/config.json');
 
-function configurePassport(authJsonContent) {
-    
-    passport.use(new OAuth2Strategy({
+function configurePassport(authJsonContent, claimNames) {
+
+    const agent = new https.Agent({
+        rejectUnauthorized: false
+    });    
+    //set scopes to call API Manager REST apis
+    const requestedScopes = "openid profile email groups roles apim:subscribe";
+    let scope = requestedScopes.split(" ");
+    scope.push(...(authJsonContent.scope ? authJsonContent.scope.split(" ") : ""));
+    const strategy = new OAuth2Strategy({
         issuer: authJsonContent.issuer,
         authorizationURL: authJsonContent.authorizationURL,
         tokenURL: authJsonContent.tokenURL,
         userInfoURL: authJsonContent.userInfoURL,
         clientID: authJsonContent.clientId,
         callbackURL: authJsonContent.callbackURL,
-        scope: authJsonContent.scope ? authJsonContent.scope.split(" ") : "",
+        scope: requestedScopes,
         passReqToCallback: true,
         state: true,
         pkce: true
     }, (req, accessToken, refreshToken, params, profile, done) => {
+        if (!accessToken) {
+            console.error('No access token received');
+            return done(new Error('Access token missing'));
+        }
         const decodedJWT = jwt.decode(params.id_token);
-        const name = decodedJWT['given_name']? decodedJWT['given_name'] : decodedJWT['nickname'];
+        const name = decodedJWT['given_name'] ? decodedJWT['given_name'] : decodedJWT['nickname'];
+        const organizationID = decodedJWT[claimNames[config.orgIDClaim]] ? decodedJWT[config.orgIDClaim] : '';
```

Comment:
Use constants instead of reading from config

Reply:
changed